aspnetcore/fundamentals/dependency-injection.md

```diff
@@ -230,131 +414,129 @@ For more information, see [Scope validation in the Web Host topic](xref:fundamen
+## Design services for dependency injection
 
-What if you find that your classes tend to have way too many dependencies being injected? This is generally a sign that your class is trying to do too much, and is probably violating SRP - the [Single Responsibility Principle](http://deviq.com/single-responsibility-principle/). See if you can refactor the class by moving some of its responsibilities into a new class. Keep in mind that your `Controller` classes should be focused on UI concerns, so business rules and data access implementation details should be kept in classes appropriate to these [separate concerns](http://deviq.com/separation-of-concerns/).
+Design services to use dependency injection to obtain their dependencies. Avoid the use of stateful static method calls (a bad practice known as [static cling](http://deviq.com/static-cling/)) and the direct instantiation of dependent classes within services. Direct instantiation "glues" the code to a particular implementation. It may help to remember the phrase, [New is Glue](https://ardalis.com/new-is-glue), when choosing whether to instantiate a type or to request it via dependency injection. By following the [SOLID Principles of Object Oriented Design](http://deviq.com/solid/), app classes naturally tend to be small, well-factored, and easily tested.
 
-With regards to data access specifically, you can inject the `DbContext` into your controllers (assuming you've added EF to the services container in `ConfigureServices`). Some developers prefer to use a repository interface to the database rather than injecting the `DbContext` directly. Using an interface to encapsulate the data access logic in one place can minimize how many places you will have to change when your database changes.
+If a class seems to have too many injected dependencies, this is generally a sign that the class has too many responsibilities and is probably violating the [Single Responsibility Principle (SRP)](http://deviq.com/single-responsibility-principle/). Attempt to refactor the class by moving some of its responsibilities into a new class. Keep in mind that Razor Pages page model classes and MVC controller classes should be focused on UI concerns. Business rules and data access implementation details should be kept in classes appropriate to these [separate concerns](http://deviq.com/separation-of-concerns/).
```

Comment:
- probably violating --> violating
- http --> https
- should be focused on --> should focus on